### PR TITLE
[highway] Add new port

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  This adds a new port for [highway](https://github.com/google/highway) (aka libhwy) 0.14.2

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Not sure yet, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/highway/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "highway",
+  "version-semver": "0.14.2",
```
Comment (chausner) [inline ports/highway/vcpkg.json:3]: Semver according to https://github.com/google/highway/blob/acc21a670582d791c6790e1f995029f57a930784/hwy/highway.h#L28.

File: ports/highway/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DBUILD_TESTING=OFF
+)
+
+vcpkg_cmake_install()
+
+# remove test-related pkg-config files that break vcpkg_fixup_pkgconfig
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/libhwy-test.pc")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/libhwy-test.pc")
```
Comment (JackBoosY) [inline ports/highway/portfile.cmake:21]: I think we should fix that instead of remove them.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/highway/portfile.cmake (hunk 2 above)
Comment (chausner) [reply ports/highway/portfile.cmake:21]: libhwy-test are only tests. I am not sure why a pkg-config file is generated for it.

The problem is that those pkg-config files list "gtest" as a pkg-config dependency.

Comment (JackBoosY) [reply ports/highway/portfile.cmake:21]: Does `libhwy.pc` exist in _lib/pkgconfig_?

Comment (chausner) [reply ports/highway/portfile.cmake:21]: Yes, there are `libhwy.pc`, `libhwy-contrib.pc` and `libhwy-test.pc` getting installed into  `lib/pkgconfig`.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**